### DISCUSSION TITLE
Why BehaviorTree V3?

### DISCUSSION THREAD

**Discussion by facontidavide** [aerostack2/aerostack2#693] [General]
### Feature Description

Hi,

I am the maintainer of Behaviortree.CPP. I have just found out that you use my library and that is great!

But I wonder if you would consider migrating to the latest version 4.X

- Branch 3.8 will not receive any update nor support, unless it is a critical bug.
- Version 4.X has many more functionalities.
- Groot has been in the state of **abandonware** for 4 years. All my effort went into developing Groot2.

If you think that upgrading is beneficial, I can help with the migration, submitting some PRs

**Comment by pariaspe** [comment]
Hi @facontidavide ! We used your library in the past when version 4.X wasn't released (amazing lib btw). We are interested for sure in upgrading to latest version (new features as XML scripting are great!). Unfortunately, we are run out of resources to do the change since we are focus on other packages of the framework right now. Anyway, it is in the roadmap and we'll try our best to upgrade asap. 
Aerostack2 is open to contribute, so feel free to submit whatever you consider. We'll be glad to review and merge your PR :)

**Comment by miferco97** [comment]
Thanks to @pariaspe’s great work, Aerostack2 now supports BT4 as of commit 68bd4b3 (merged in #839)!
Hopefully this migration inspires more planning-related contributions and helps more people adopt Aerostack2 for their aerial robotics projects 🤖🚁
